osu.Game.Rulesets.Mania.Tests/TestSceneHoldNoteInput.cs

```diff
@@ -388,7 +388,7 @@ private void performTest(List<ReplayFrame> frames, Beatmap<ManiaHitObject> beatm
                     },
                 };
 
-                beatmap.ControlPointInfo.Add(0, new DifficultyControlPoint { SpeedMultiplier = 0.1f });
+                beatmap.ControlPointInfo.Add(0, new DifficultyControlPoint { SliderVelocity = 0.1f });
```

Comment:
Yep, for sure. Some of these tests are unfortunately visual only, and as per my own admission I haven't gone through and tested all scenarios yet.

Reply:
I'm doing a second pass on this be actually throwing when a control point is attempted to be added to a `ControlPointInfo` that doesn't handle its type. Should hopefully catch all these cases (and has initially surfaced quite a few new cases to check on).

Reply:
@bdach can you explain the `DistanceObjectPatternGenerator` one? Is there test coverage of this somewhere i'm missing, or do you mean the usage in the class itself? If the second, I *think* that's actually required to get some weird precision from the special legacy point type.

Reply:
Well I mean it seems a bit weird that almost *all* of mania changes in this diff use `ScrollSpeed` except for this one case that uses `SliderVelocity`. Especially so that `ManiaBeatmapConverter` converts slider velocity adjustments and difficulty control points to scroll speed adjustments and effect control points in the ctor, and that happens before the pattern generators are invoked I'm pretty sure...

`ManiaBeatmapConverterTest` exists but from a quick look I don't think it tests nonstandard scroll speeds in the context of the stair pattern generation.

The precision stuff is there but it may be required to make the `LegacyDifficultyControlPoint` a `LegacyEffectControlPoint` now?

Reply:
For sure. Going to go over this with a fine comb. I was trying to avoid changing/dealing with this because I still don't even fully understand it, but let's see how things go..

Reply:
So following the flow, this generation pass is done *after* the conversion of `DifficultyControlPoint`s to `EffectControlPoint`s... but the conversion is only done for osu!mania specific beatmaps, which would mean there is no object conversion.

So in a round-about convoluted way, this is actually correct as it stands. Not sure how I feel about that in terms of being able to understand everything going on. @smoogipoo can you chime in on this one when you get to reviewing this PR?

Reply:
Oh, right, that explains the difference in handling. I'd personally be ok with an inline comment in that case.